### ISSUE TITLE
Garbled SourceView for some ghost zones

### ISSUE THREAD

**Issue by bjacob** [wolfpld/tracy#379]
I have big traces with sampling (ghost zones).  About 3/4 of the ghost zones load up fine in the TracySourceView but the remaining ~ 1/4 of ghost zones give a garbled SourceView.  In that case, both the left pane (source) and right pane (disassembly) are garbled.

Actually for a tiny fraction of a second initially, the TracySourceView seems like it might be fine (hard to be sure - this just flickers for maybe ~ 1 frame).

After that initial flicker, the view shows as if all of the source/asm lines had been painted over the first line or two:

![image](https://user-images.githubusercontent.com/79535/166562176-cb924b03-9448-4728-b629-184fde91d9c0.png)

If I scroll down, some stuff happens - but still garbled:

![image](https://user-images.githubusercontent.com/79535/166562360-b6b019b1-1094-4d9c-8fda-17afbd85caba.png)

Eventually I triggered this assertion failure - not sure if related:
```
iree-tracy-profiler: /usr/local/google/home/benoitjacob/iree/third_party/tracy/server/TracySourceView.cpp:4435: void tracy::SourceView::SelectAsmLinesHover(uint32_t, uint32_t, const tracy::Worker &): Assertion `m_selectedAddressesHover.empty()' failed.
```

`iree-tracy-profiler` is just the regular tracy profiler -- just with our own CMakeLists.txt.

**Comment by wolfpld** [comment]
The assertion is due to the line-hover logic expecting a clean state, which it obviously doesn't have, if the hover is activated multiple times due to line overlaying.

Which version of Tracy are you using here?

**Comment by bjacob** [comment]
That was with today's `master`: `f4581c398c19df34aac0bc15716e1a9426ba7736`

That happens also with the commit that IREE is currently sync'd at, which is Tracy 0.8.1: `80f6a93da79af94fc4604e37439f5a75d5e7dfbd`.

I'm on Debian Linux, x86-64.

**Comment by bjacob** [comment]
Sorry, that was a bug that I introduced in #369 , not understanding that `ImGui::SameLine` actually walks back to the *previous* line so isn't idempotent. Fixed in #381.
